File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for i in range(len(nums)):
            dif = target - nums[i]

            if dif in seen:
                return [i, seen.get(dif)]
            
            seen[nums[i]] = i

        
    
    # SC -> O(N)
    # TC -> O(N)




        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> lookUp = new HashMap<>();

        for(int i = 0; i < nums.length;i++){
            int compNum = target - nums[i];
            if(lookUp.containsKey(compNum)){
                return new int[] {lookUp.get(compNum),i};
            }

            lookUp.put(nums[i],i); 
        }

        return new int[] {};
        
    }
}

// SC -> O(N)
// TC -> O(N)